trame_vtk/modules/vtk/addon_serializer.py

```diff
@@ -64,6 +65,213 @@ def extractRequiredFields(
+            )
+
+    if dataObjectInstance:
+        colorArrayName = (
+            mapper.GetArrayName()
+            if mapper.GetArrayAccessMode() == 1
+            else mapper.GetArrayId()
+        )
+        return {
+            "parent": render_window_serializer.getReferenceId(parent),
+            "id": mapperId,
+            "type": render_window_serializer.class_name(mapper),
+            "properties": {
+                "resolveCoincidentTopology": mapper.GetResolveCoincidentTopology(),
+                "renderTime": mapper.GetRenderTime(),
+                "arrayAccessMode": mapper.GetArrayAccessMode(),
+                "scalarRange": mapper.GetScalarRange(),
+                "useLookupTableScalarRange": 1
+                if mapper.GetUseLookupTableScalarRange()
+                else 0,
+                "scalarVisibility": mapper.GetScalarVisibility(),
+                "colorByArrayName": colorArrayName,
+                "colorMode": mapper.GetColorMode(),
+                "scalarMode": mapper.GetScalarMode(),
+                "interpolateScalarsBeforeMapping": 1
+                if mapper.GetInterpolateScalarsBeforeMapping()
+                else 0,
+            },
+            "calls": calls,
+            "dependencies": dependencies,
+        }
+
+    return None
+
+
+def scalarBarActorSerializer(parent, actor, actorId, context, depth):
+    dependencies = []
+    calls = []
+    lut = actor.GetLookupTable()
+    if not lut:
+        return None
+
+    lutId = render_window_serializer.getReferenceId(lut)
+    lutInstance = render_window_serializer.serializeInstance(
+        actor, lut, lutId, context, depth + 1
+    )
+    if not lutInstance:
+        return None
+
+    dependencies.append(lutInstance)
+    calls.append(["setScalarsToColors", [render_window_serializer.wrapId(lutId)]])
+
+    prop = None
+    if hasattr(actor, "GetProperty"):
+        prop = actor.GetProperty()
+    else:
+        if context.debugAll:
+            print("This scalarBarActor does not have a GetProperty method")
+
+        if prop:
+            propId = render_window_serializer.getReferenceId(prop)
+            propertyInstance = render_window_serializer.serializeInstance(
+                actor, prop, propId, context, depth + 1
+            )
+            if propertyInstance:
+                dependencies.append(propertyInstance)
+                calls.append(["setProperty", [render_window_serializer.wrapId(propId)]])
+
+    axisLabel = actor.GetTitle()
+    width = actor.GetWidth()
+    height = actor.GetHeight()
+
+    # axisTitlePixelOffset = actor.GetTextPad()
+    # position = actor.GetPosition()
+    # position2 = actor.GetPosition2()
+    # print(f'axisTitlePixelOffset: {axisTitlePixelOffset}')
+    # print(f'axisLabel: {axisLabel}')
+    # print(f'width: {width}')
+    # print(f'height: {height}')
+    # print(f'position: {position}')
+    # print(f'position2: {position2}')
+
```

Comment:
Should we remove all those commented lines?

Reply:
I left to be consistent with the method it's patching...

I agree they should be removed. I'll clean this up here then after open a PR in VTK